docs/enhancements/fep-001-api-fleet-templating.md

```diff
@@ -0,0 +1,257 @@
+# EP: Definition of the Fleet and Device API spec
+<!-- this format is inspired by the K8S KEP format https://raw.githubusercontent.com/kubernetes/enhancements/master/keps/NNNN-kep-template/README.md -->
+## Summary
+
+The Fleet API is designed to manage big groups of devices that share configuration
+and operating system image.
+
+Some use cases point to the necessity of customizing some of those details based
+on the device's metadata.
+
+## Motivation
+
+Allowing templating of the device spec in the fleet API we could achieve a more
+flexible management, allowing customers to manage devices in a more granular way.
+
+### Goals
+
+* Define templating mechanism for the device spec template configuration in fleets,
+  based on labels and other metadata.
+
+* Enable different os images based on the device images.
+```yaml
+  template:
+    spec:
+      os:
+        image: access-control-{{ .device.labels["region"] }}:1.0.0
+```
+
+* Enable different references for a config source based on labels: 
+```yaml
+- name: microshift-manifests
+  configType: GitConfigProviderSpec
+  gitRef:
+    repository: basic-nginx-demo
+    targetRevision: 1.0.0-{{ .device.labels["region"] }}
+    path: /basic-nginx-demo/configuration
```

Comment:
From a UX perspective do we need a dry-run or such ability to validate these templates client side?